app_sensor_module/src/sensors.c

```diff
@@ -0,0 +1,164 @@
+                telemetry->lis3mdl.mag_y = sensor_values[1];
+                telemetry->lis3mdl.mag_z = sensor_values[2];
+                break;
+            case SENSOR_MODULE_MS5611:
+                l_get_sensor_data_float(sensors[i], 2, channels_arr[i], sensor_values);
+                telemetry->ms5611.pressure = sensor_values[0];
+                telemetry->ms5611.temperature = sensor_values[1];
+                break;
+            case SENSOR_MODULE_BMP388:
+                l_get_sensor_data_float(sensors[i], 2, channels_arr[i], sensor_values);
+                telemetry->bmp388.pressure = sensor_values[0];
+                telemetry->bmp388.temperature = sensor_values[1];
+                break;
+            default:
+                LOG_ERR("Invalid sensor index");
+                break;
+        }
+    }
+}
+
+static void hundred_hz_sensor_reading_task(void *unused0, void *unused1, void *unused2) {
+    ARG_UNUSED(unused0);
+    ARG_UNUSED(unused1);
+    ARG_UNUSED(unused2);
+
+    // Initialize variables for receiving telemetry
+    sensor_module_hundred_hz_telemetry_t hundred_hz_telemetry;
+    uint32_t timestamp = 0;
+
+    const struct device *sensors[SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS] = {
+            [SENSOR_MODULE_ADXL375] = DEVICE_DT_GET(DT_INST(0, adi_adxl375)),
+            [SENSOR_MODULE_MS5611]  = DEVICE_DT_GET(DT_INST(0, meas_ms5611)),
+            [SENSOR_MODULE_BMP388]  = DEVICE_DT_GET(DT_INST(0, bosch_bmp388)),
+            [SENSOR_MODULE_LSM6DSL] = DEVICE_DT_GET(DT_INST(0, st_lsm6dsl)),
+            [SENSOR_MODULE_LIS3MDL] = DEVICE_DT_GET(DT_INST(0, st_lis3mdl_magn))
+    };
+
+    const enum sensor_channel *channels_arr[SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS] = {
+            [SENSOR_MODULE_ADXL375] = accelerometer_channels,
+            [SENSOR_MODULE_MS5611]  = barometer_channels,
+            [SENSOR_MODULE_BMP388]  = barometer_channels,
+            [SENSOR_MODULE_LSM6DSL] = imu_channels,
+            [SENSOR_MODULE_LIS3MDL] = magnetometer_channels
+    };
+
+    // Confirm sensors are ready
+    bool sensor_ready[SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS] = {false};
+    check_sensors_ready(sensors, sensor_ready, SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS);
+
+    while (true) {
+        // Refresh sensor data
+        l_update_sensors_safe(sensors, SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS, sensor_ready);
+        timestamp = k_uptime_get_32();
+
+        // Populate hundred_hz_telemetry with refreshed data. Need to manually assign due to packing.
+        populate_telemetry_struct(&hundred_hz_telemetry, sensors, channels_arr);
+
+        // Put telemetry into queue
+        if (k_msgq_put(&hundred_hz_telem_queue, &hundred_hz_telemetry, K_NO_WAIT)) {
+            LOG_ERR("Failed to put data into INA219 processing queue");
+        }
+
+        // Sleep until next update time. TODO: Maybe use timers and signals?
+        uint32_t time_to_wait = HUNDRED_HZ_UPDATE_TIME - (k_uptime_get_32() - timestamp);
```

Comment:
Would be good to upgrade this to a timer. Its pretty trivial.

Reply:
Would also need to mess with signals. Made this a TODO since we should update power mod code to be the same as whatever mechanism we are using. Just doing the minimum to get us in a working state for now